### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by Eccb7**
**Summary by @Eccb7 and @kkmanuu :**

This pull request addresses the following issues and introduces the following changes:

- [Issue #1](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-all-3F): This issue was related to the implementation of the `all?` method in the `MyEnumerable` module. It is resolved by adding proper logic and tests.
- [Issue #2](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-any-3F): This issue involved the `any?` method in the `MyEnumerable` module. We've improved the method implementation and added new test cases.
- [Issue #3](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-filter): The `filter` method needed enhancements to handle edge cases. We've made improvements and expanded test coverage.

**Changes Made:**

In this PR, we have made the following key changes and improvements:

- Enhanced the logic and implementation of `all?`, `any?`, and `filter` methods in the `MyEnumerable` module.
- Added comprehensive test cases to ensure these methods work as expected.
- Refactored the code for readability and adherence to coding standards.

These changes are necessary to ensure the correct functionality of the Enumerable methods and to maintain code quality.

**Checklist:**

- [x] Code follows the project's style and coding guidelines.
- [x] Documentation (README, comments) is updated to reflect the changes.
- [x] Commit messages are clear and follow the project's commit message conventions.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class MyList
```
Comment (t-yanick) [inline my_list.rb:3]: - [ ] Great job working on this milestone. You should use `require_relative` to import `MyEnumerable` since using `require_relative` for files in your directory is a ruby convention.  `require_relative` is used for referring to files within your directory.
`require_relative 'my_enumerable'`

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Eccb7 and @kkmanuu , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Two separate files have been used
:heavy_check_mark: `MyList` class has the `#initialize` method that accepts multiple arguments.
:heavy_check_mark: Collaborative work 👍Commits from both team members.
:heavy_check_mark: `MyEnumerable` module implements `#all?` `#any?` and `#filter` method

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Eccb7, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
